src/main/java/com/algosenpai/app/ui/Ui.java

```diff
@@ -89,10 +90,21 @@ public Ui() throws FileNotFoundException {
     @FXML
     public void initialize() {
         scrollPane.vvalueProperty().bind(dialogContainer.heightProperty());
-        dialogContainer.getChildren().add(DialogBox.getSenpaiDialog(GREETING_MESSAGE, senpaiImage));
-        userPic.setImage(userImage);
-        levelProgress.setProgress(0);
-        playerLevel.setText("You are Level 1");
+        UserStats tempStats = UserStats.parseString(Storage.loadData("UserData.txt"));
+        if (tempStats.getUsername().equals("DefaultName")) {
+            dialogContainer.getChildren().add(DialogBox.getSenpaiDialog(GREETING_MESSAGE, senpaiImage));
+            userPic.setImage(userImage);
+            levelProgress.setProgress(0);
+            playerLevel.setText("You are Level 1");
+        } else {
+            dialogContainer.getChildren().add(DialogBox.getSenpaiDialog(
+                    "Welcome back " + tempStats.getUsername(), senpaiImage));
+            userPic.setImage(userImage);
+            levelProgress.setProgress(0);
+            playerLevel.setText("You are Level " + tempStats.getUserLevel());
+        }
+
+
         handle();
```

Comment:
on initialize, display the`GREETING_MESSAGE` only if the stats name is "DefaultName" , indicating that the user has launched the program for the first time

Else skip that and just say welcome + name